arrow2 0.18.0 release broke against minimal dependencies in the Cargo.toml

Even though #1568 is not merged in, apparently version 0.18.0 has already been released to crates.io. In that release, you are putting a dependency on an API introduced in chrono 0.4.31, but in your Cargo.toml, you only have a dependency on 0.4. In my application, since I am using a Cargo.lock file, the upgrade to arrow2 0.18.0 caused a failure since I have an older version of chrono than 0.4.31. Bumping the actual dependency to 0.4.31 in your Cargo.toml would require a new enough version of chrono to not break builds.

Same here. I could fix it by upgrading my `chrono` version, but it was bit of a nuisance to debug. I had `chrono` and `polars` as dependencies, and not `arrow2` directly.

In case someone is googling around, the error was:
```rust 
error[E0599]: no method named `timestamp_nanos_opt` found for struct `DateTime` in the current scope
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,7 +20,7 @@ either = "1.9"
 num-traits = "0.2"
 dyn-clone = "1"
 bytemuck = { version = "1", features = ["derive"] }
-chrono = { version = "0.4", default_features = false, features = ["std"] }
+chrono = { version = "0.4.31", default_features = false, features = ["std"] }
 
 # for decimal i256
 ethnum = "1"

